addons/purchase_stock/tests/test_reordering_rule.py

```diff
@@ -297,6 +297,133 @@ def test_reordering_rule_4(self):
                 'product_max_qty': 1,
             })
 
+    def test_reordering_rule_best_vendor_price_selection(self):
+        """ Check that when multiple vendor price rules exist for multiple vendors,
+        the reordering rule always applies the correct vendor and vendor price rule
+        depending on the ordered quantity and selected Vendor.
+
+        - Create a product with two vendors:
+            Vendor 1 → (1.0 Units - $1.00)
+            Vendor 2 → (1.0 Units - $10.00), (5.0 Units - $5.00)
+        - Create a reordering rule for the product (without setting vendor)
+        - Trigger replenishment with qty = 2 → should pick Vendor 1's $1.00 price rule
+        - Cancel the generated PO
+        - Create a reordering rule for the product (using Vendor V2)
+        - Trigger replenishment with qty = 4 → should pick Vendor V2's $10.00 price rule
+        - Cancel the generated PO
+        - Trigger replenishment again with qty = 5 → should pick Vendor V2's $5.00 price rule
+        """
+        self.env.user.group_ids |= self.env.ref('stock.group_stock_multi_locations')
+        partner1, partner2 = self.env['res.partner'].create([
+            {'name': 'Vendor V1'},
+            {'name': 'Vendor V2'},
+        ])
+        warehouse = self.env['stock.warehouse'].search([('company_id', '=', self.env.company.id)], limit=1)
+        uom_unit = self.env.ref('uom.product_uom_unit')
+
+        product_form = Form(self.env['product.product'])
+        product_form.name = 'Product A'
+        product_form.is_storable = True
+        price_rules = [
+            {'partner': partner1, 'min_qty': 1.0, 'price': 1.0},
+            {'partner': partner2, 'min_qty': 1.0, 'price': 10.0},
+            {'partner': partner2, 'min_qty': 5.0, 'price': 5.0},
+        ]
+        for rule in price_rules:
+            with product_form.seller_ids.new() as s:
+                s.partner_id = rule['partner']
+                s.min_qty = rule['min_qty']
+                s.price = rule['price']
+                s.product_uom_id = uom_unit
+        product = product_form.save()
+
+        # Reordering rule without vendor → should pick Vendor V1 → 2 @ $1
+        orderpoint_form = Form(self.env['stock.warehouse.orderpoint'])
+        orderpoint_form.warehouse_id = warehouse
+        orderpoint_form.location_id = warehouse.lot_stock_id
+        orderpoint_form.product_id = product
+        order_point = orderpoint_form.save()
+
+        order_point.product_min_qty = 2.0
+        order_point.product_max_qty = 2.0
+        order_point.action_replenish()
+
+        po = self.env['purchase.order'].search([('product_id', '=', product.id)])
+        pol = po.order_line
```

Comment:
```suggestion
        pol = self.env['purchase.order'].search([('product_id', '=', product.id)]).order_line
```
Maintain consistency everywhere.